File: source/205.cpp
bool isIsomorphic(string s, string t) {
	int count_s = 1, count_t = 1, cur_s, cur_t;
	map<char, int> hash_s, hash_t;
	for (int index = 0; index < s.length(); ++index) {
		if (hash_s.count(s.at(index)) == 0) hash_s[s.at(index)] = count_s++;
		cur_s = hash_s[s.at(index)];
		if (hash_t.count(t.at(index)) == 0) hash_t[t.at(index)] = count_t++;
		cur_t = hash_t[t.at(index)];
		if (cur_s != cur_t) return false;
	}
	return true;
}

File: sourcePY/205.py
class Solution:
    def isIsomorphic(self, s, t):
        count_s, count_t = 0, 0
        dict_s, dict_t = {}, {}
        for index in range(len(s)):
            if not s[index] in dict_s:
                dict_s[s[index]] = count_s
                count_s += 1
            cur_s = dict_s[s[index]]
            if not t[index] in dict_t:
                dict_t[t[index]] = count_t
                count_t += 1
            cur_t = dict_t[t[index]]
            if not cur_s == cur_t:
                return False
        return True